cmd/api.go

```diff
@@ -125,9 +124,6 @@ func (o *apiOptions) bindFlags(cmd *cobra.Command) {
 			"generates an API following an extension pattern (addon)")
 	}
 
-	cmd.Flags().BoolVar(&o.force, "force", false,
```

Comment:
Dropping a flag is definitely a breaking change.
If we really want to do this, we need to start to deprecate it in v2 and drop it in v3.

Reply:
And we would need a better way to track both external resources and already created CRDs.

Reply:
HI @mengqiy, 

Thank you for your input. Regards your comment, could you please answer my following questions to help me understand how it works? 

> Dropping a flag is definitely a breaking change.
> If we really want to do this, we need to start to deprecate it in v2 and drop it in v3.

Was not thee V1 / V2 done just because of the break changes in the layout of the scaffolded project and in order to still supporting the projects made in the old and new layout? Do we need to create a new structure/layout to scaffold files besides it has no changes at all?   Can we not just release a new tag version 3.0.0 since the breaking change is just in the command?

Regards the change/solution to fix. I will see if we can achieve the same goal without the need to remove the flag.

Reply:
Hi @Adirio and @mengqiy. 

Could you please check the changes made. We can solve the same scenario whiteout a need to introduce a break change in the command tool.

Reply:
**Disclaimer:** the following comment only focuses on whether a flag change should require a amjor version bump or not, as they issue solved by this PR has been considered NOT to be a bug: https://github.com/kubernetes-sigs/kubebuilder/issues/1263#issuecomment-591113294

The kubebuilder version/release (currently `v2.2.0`) vs scaffold version (currently `"2"`) issue has been appearing in several places now. As you have said, removing a flag does not affect the scaffolded files, so the scaffold version could stay as is, but it does affect the CLI, which would require a major version bump for the kubebuilder version/release to `v3.0.0`.

Implementing the plugin system will make the difference clearer, but a third one is going to appear, plugin version/release. Once plugins are implemented, flags belong to plugins as it can be seen in the code fragment below, so the plugin version/release is the one that should be updated.
https://github.com/kubernetes-sigs/kubebuilder/blob/ebd0d8c99aa648778ce6598146667f13b717306e/pkg/plugin/v2/api.go#L94-L116